pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuStateVec_helpers.hpp

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022-2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Adapted from JET: https://github.com/XanaduAI/jet.git
+
+/**
+ * @file cudaStateVec_helpers.hpp
+ */
+
+#pragma once
+#include <custatevec.h>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "cuStateVecError.hpp"
+
+namespace Pennylane::LightningGPU::Util {
+
+inline static auto pauliStringToEnum(const std::string &pauli_word)
+    -> std::vector<custatevecPauli_t> {
+    // Map string rep to Pauli enums
+    const std::unordered_map<std::string, custatevecPauli_t> pauli_map{
+        std::pair<const std::string, custatevecPauli_t>{std::string("X"),
+                                                        CUSTATEVEC_PAULI_X},
+        std::pair<const std::string, custatevecPauli_t>{std::string("Y"),
+                                                        CUSTATEVEC_PAULI_Y},
+        std::pair<const std::string, custatevecPauli_t>{std::string("Z"),
+                                                        CUSTATEVEC_PAULI_Z},
+        std::pair<const std::string, custatevecPauli_t>{std::string("I"),
+                                                        CUSTATEVEC_PAULI_I}};
+
+    static constexpr std::size_t num_char = 1;
```

Comment:
does it need to be static? :thinking: I may be missing something but I guess `constexpr` is sufficient here.

```suggestion
    constexpr std::size_t num_char = 1;
```